clients/src/main/java/org/apache/kafka/clients/consumer/internals/events/ConsumerRequestEvent.java

```diff
@@ -0,0 +1,20 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.clients.consumer.internals.events;
+
+abstract public class ConsumerRequestEvent {
```

Comment:
Can you clarify what qualifies as a "request event" vs "response event"?

Reply:
I'll add java docs around it.

Reply:
Per my other comment above: for request, do we also want to always include a future?

Reply:
Probably not, events like assignment don't really require a response.